zigpy/zcl/__init__.py

```diff
@@ -374,7 +374,14 @@ def _write_attr_records(
     async def write_attributes(
         self, attributes: Dict[Union[str, int], Any], manufacturer: Optional[int] = None
     ) -> List:
+        """Write attributes to device with internal 'attributes' validation"""
         args = self._write_attr_records(attributes)
+        return self.write_attributes_raw(args, manufacturer)
+
+    async def write_attributes_raw(
+        self, args: List[foundation.Attribute], manufacturer: Optional[int] = None
```

Comment:
`args` was an internal variable name but making it a parameter necessitates a more useful name.  Could you rename it to `attrs`?

Reply:
I renamed it, but I have to review the test cases I added.